### ISSUE TITLE
FailedRenderFileException

### ISSUE THREAD

**Issue by vlastavesely** [ApiGen/ApiGen#890]
Hi,
I've just installed in the way you suggested in README. I tried to run it on a test class I made.

```php
<?php

use Nette;

final class Lorem
{
	use Nette\SmartObject;

	/**
	 * @param string    used for nothing
	 * @param int|NULL  also used for nothing
	 * @return void
	 */
	public function test(string $param1, ?int $param2)
	{
		$this->someCode();
	}
}
```

But it throws an exception:

```
[ApiGen\Exception\Templating\FailedRenderFileException]
Rendering of "/var/www/html/apigen/vendor/apigen/apigen/packages/ModularConfiguration/src/Option/../../../../packages/ThemeDefault/src/class.latte" template to "/var/www/html/apigen/docs/class-Lorem.html" file failed due to: Argument for filter "linkReflection" has to be type of "ApiGen\Reflection\Contract\Reflection\AbstractReflectionInterface". "string" given.
```

Problem is in `use Nette\SmartObject` line. When it is removed, everything works. I checked whether `nette\utils` is installed and it is. Sadly, I do not have an insight into ApiGen's internals so I do not know where could be the problem.

**Comment by vlastavesely** [comment]
The same problem is on my main project, on which I'm gonna test Apigen continuously.

**Comment by TomasVotruba** [comment]
Hey, thanks for report.

Could you send failing test for this?

Just:

1. copy paste [ClassGeneratorTest](https://github.com/ApiGen/ApiGen/blob/f1c3b383453daf63b6d62ffe4be8c0d27e058828/tests/Generator/ClassGeneratorTest.php#L20) - name it like ClassWithTraitGeneratorTest,
2. create own `/*Source` directory with the code you provided
3. and [expect generated file to exist](https://github.com/ApiGen/ApiGen/blob/f1c3b383453daf63b6d62ffe4be8c0d27e058828/tests/Generator/ClassGeneratorTest.php#L30).

**Comment by vlastavesely** [comment]
ok

**Comment by TomasVotruba** [comment]
Thanks for the PR, it helped me to find bug very fast.

Fixed via: #893 
This  commit in particular: https://github.com/ApiGen/ApiGen/pull/893/commits/a9213b9796233fd9856a291102ba8d94b1c9b88e

```html
-        <tr><th>Methods used from <a href="{$traitName|linkReflection}#methods">{$traitName}</a> trait</th></tr>
+        <tr><th>Methods used from <a href="{$trait|linkReflection}#methods">{$traitName}</a> trait</th></tr>
```
